src/Polly.Core/Utils/Pipeline/ExecutionTrackingComponent.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace Polly.Utils.Pipeline;
+
+internal sealed class ExecutionTrackingComponent : PipelineComponent
+{
+    public static readonly TimeSpan Timeout = TimeSpan.FromSeconds(30);
+
+    public static readonly TimeSpan SleepDelay = TimeSpan.FromSeconds(1);
+
+    private readonly TimeProvider _timeProvider;
+    private int _pendingExecutions;
```

Comment:
Worth making a `long`?

Reply:
I don't think any strategy will have that many pending executions.